src/code/z_actor.c

```diff
@@ -457,439 +1246,3471 @@ f32 Actor_XZDistanceToPoint(Actor* actor, Vec3f* point) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+
+    idx = Quake_Add(&globalCtx->mainCamera, 3);
+    Quake_SetSpeed(idx, speed);
+    Quake_SetQuakeValues(idx, arg1, 0, 0, 0);
+    Quake_SetCountdown(idx, arg2);
+}
+
+void func_800BC848(Actor* actor, GameState* gameState, s16 arg2, s16 arg3) {
+    if (arg2 >= 5) {
+        func_8013ECE0(actor->xyzDistToPlayerSq, 0xFF, 0x14, 0x96);
+    } else {
+        func_8013ECE0(actor->xyzDistToPlayerSq, 0xB4, 0x14, 0x64);
+    }
+    func_800BC770(gameState, arg2, arg3);
+}
+
+DoorLockInfo sDoorLocksInfo[DOORLOCK_MAX] = {
+    /* DOORLOCK_NORMAL */ { 0.54f, 6000.0f, 5000.0, 1.0f, 0.0f, D_05000230, D_05000140 },
+    /* DOORLOCK_BOSS */ { 0.644f, 12000.0f, 8000.0f, 1.0f, 0.0f, D_06000530, D_06000400 },
+    /* DOORLOCK_2 */ { 0.6400000453f, 8500.0f, 8000.0f, 1.75f, 0.1f, D_05000230, D_05000140 },
+};
+
+/**
+ * Draws chains and lock of a locked door, of the specified `type` (see `DoorLockType`).
+ * `frame` can be 0 to 10, where 0 is "open" and 10 is "closed", the chains slide accordingly.
+ */
+void Actor_DrawDoorLock(GlobalContext* globalCtx, s32 frame, s32 type) {
+    s32 pad[2];
+    MtxF baseMtxF;
+    s32 i;
+    f32 chainsTranslateX;
+    f32 chainsTranslateY;
+    DoorLockInfo* entry = &sDoorLocksInfo[type];
+    f32 chainRotZ = entry->chainsRotZInit;
+    f32 rotZStep;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    Matrix_InsertTranslation(0.0f, entry->yShift, 500.0f, MTXMODE_APPLY);
+    Matrix_CopyCurrentState(&baseMtxF);
+
+    chainsTranslateX = __sinf(entry->chainAngle - chainRotZ) * -(10 - frame) * 0.1f * entry->chainLength;
+    chainsTranslateY = __cosf(entry->chainAngle - chainRotZ) * (10 - frame) * 0.1f * entry->chainLength;
+
+    for (i = 0; i < 4; i++) {
+        Matrix_SetCurrentState(&baseMtxF);
+        Matrix_InsertZRotation_f(chainRotZ, MTXMODE_APPLY);
+        Matrix_InsertTranslation(chainsTranslateX, chainsTranslateY, 0.0f, MTXMODE_APPLY);
+        if (entry->chainsScale != 1.0f) {
+            Matrix_Scale(entry->chainsScale, entry->chainsScale, entry->chainsScale, MTXMODE_APPLY);
+        }
+
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_OPA_DISP++, entry->chainDL);
+
+        if ((i % 2) != 0) {
+            rotZStep = 2.0f * entry->chainAngle;
+        } else {
+            rotZStep = M_PI - (2.0f * entry->chainAngle);
+        }
+
+        chainRotZ += rotZStep;
+    }
+
+    Matrix_SetCurrentState(&baseMtxF);
+    Matrix_Scale(frame * 0.1f, frame * 0.1f, frame * 0.1f, MTXMODE_APPLY);
+
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, entry->lockDL);
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void Actor_SpawnShieldParticlesMetal(GlobalContext* globalCtx, Vec3f* v) {
+    CollisionCheck_SpawnShieldParticlesMetal(globalCtx, v);
+}
```

Comment:
```suggestion
void Actor_SpawnShieldParticlesMetal(GlobalContext* globalCtx, Vec3f* pos) {
    CollisionCheck_SpawnShieldParticlesMetal(globalCtx, pos);
}
```
I followed the trail of functions and this is actually taking in a positional argument. So the name should reflect that